manifests/agent.pp

```diff
@@ -249,6 +252,7 @@
   Variant[String[1],Array[String[1]]] $zbx_groups = $zabbix::params::agent_zbx_groups,
   $zbx_group_create                               = $zabbix::params::agent_zbx_group_create,
   $zbx_templates                                  = $zabbix::params::agent_zbx_templates,
+  Optional[Array[Hash]] $zbx_macros               = $zabbix::params::agent_zbx_macros,
```

Comment:
for arrays we try to use `[]` as default, or `{}` for hashes. In that case we won't need the `Optional` part. I think that would be possible here was well?

We've more of our review guidelines available at https://voxpupuli.org/docs/reviewing_pr/